src/eval.rs

```diff
@@ -310,10 +310,14 @@ pub fn eval_entry<'tcx>(
         }
     };
 
+    let info = ecx.preprocess_diagnostics();
```

Comment:
This should probably be called *before* the `main` stack frame is pushed. After all the diagnostics are not associated with main but with the runtime "above" it.

Reply:
If we call it early enough, we might even be *sure* that no diagnostics have been pushed yet, so we can keep the assertion where it was?

Also, please add a comment.